### PR TITLE
Optimize out some redundant code or calls to "slow" functions

### PR DESCRIPTION
**Pull request by nthykier**

### CODE REVIEW

File: scour/scour.py [hunk 1]
```diff
@@ -3627,8 +3625,8 @@ def scourString(in_string, options=None):
     # on the first pass, so we do it multiple times
     # does it have to do with removal of children affecting the childlist?
     if options.keep_editor_data is False:
-        while removeNamespacedElements(doc.documentElement, unwanted_ns) > 0:
-            pass
+        _num_elements_removed += removeNamespacedElements(doc.documentElement,
```
Comment (Ede123) [inline scour/scour.py:3631]: Would be great if this were true, but someone added the comment above for some reason at some point, so I wonder if the loop can really be removed?

File: scour/scour.py [hunk 2]
```diff
@@ -3631,8 +3629,8 @@ def scourString(in_string, options=None):
     if options.keep_editor_data is False:
         while removeNamespacedElements(doc.documentElement, unwanted_ns) > 0:
             pass
-        while removeNamespacedAttributes(doc.documentElement, unwanted_ns) > 0:
-            pass
+        _num_attributes_removed += removeNamespacedAttributes(doc.documentElement,
```
Comment (Ede123) [inline scour/scour.py:3633]: dito

File: scour/scour.py (hunk 1 above)
Comment (nthykier) [reply scour/scour.py:3631]: Honestly, I cannot see how that comment can still be true.  Unfortunately, both the functions and the comments predates the version history in git, so it is not possible to tell if it underlying problem was fixed and the comments were just left there to rot.

But if the comments are true, then we are missing a regression test case to show the issue.  Sadly, it is a bit of a devil's proof and my head is convinced that there is no such case (doesn't make it true; I cannot find a case but my thinking is biased).

I am fine with ripping those two commits out of the branch if you are not comfortable with them.

File: scour/scour.py (hunk 2 above)
Comment (nthykier) [reply scour/scour.py:3633]: Ditto as well :)

File: scour/scour.py [hunk 3]
```diff
@@ -74,6 +74,9 @@
 COPYRIGHT = u'Copyright Jeff Schiller, Louis Simard, 2010'
 
 
+# Used to split values where "x y" or "x,y" or a mix of the two is allowed
+SPACE_COMMA_SEPARATOR = re.compile(r"\s*[\s,]\s*")
```
[the remark this answers is not in the document]
Comment (nthykier) [reply scour/scour.py]: Done

File: scour/scour.py (hunk 1 above)
Comment (Ede123) [reply scour/scour.py:3656]: I think you're right, let's keep the commits and remove the comment!

Note you can [get full history](https://github.com/scour-project/scour/issues/23#issuecomment-241896530) with `git fetch origin 'refs/replace/*:refs/replace/*'` or via [old bzr history](https://bazaar.launchpad.net/~scouring/scour/trunk/changes) on Launchpad.

Looking at that, the comment was added in https://github.com/scour-project/scour/commit/879300373f39319c76ef58bd10ce0e32ee55d22b and fixed already in https://github.com/scour-project/scour/commit/2dc788aa3f5196a14c00262238ba838248a48077.

**Pull request merged**